src/ui/popup.js

```diff
@@ -512,6 +522,47 @@ export default class Popup extends Evented {
         this._update(event.point);
     }
 
+    _getAnchor(offset: any) {
+        if (this.options.anchor) { return this.options.anchor; }
+
+        const pos: any = this._pos;
+        const width = this._container.offsetWidth;
+        const height = this._container.offsetHeight;
+
+        let anchorComponents;
+        if (pos.y + offset.bottom.y < height) {
+            anchorComponents = ['top'];
+        } else if (pos.y > this._map.transform.height - height) {
```

Comment:
What does it mean when `this._map.transform.height` does not equal `_container.offsetHeight`?

Reply:
I'm not sure what you're getting at here. `this._map.transform.height` is the height of the map, while `_container.offsetHeight` is the height of the popup.

Reply:
Ohhhh sorry. I was thinking map container element. 🤦